src/power_grid_model_io/data_stores/csv_dir_store.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+CSV Directory Store
+"""
+
+from pathlib import Path
+from typing import Any, Callable, Dict, List
+
+import pandas as pd
+
+from power_grid_model_io.data_stores.base_data_store import BaseDataStore
+from power_grid_model_io.data_types import TabularData
+
+
+class CsvDirStore(BaseDataStore[TabularData]):
+    """
+    CSV Directory Store
+
+    The first row of each .csv file is expected to contain the column names, unless specified differently by an
+    extension of this class.
+    """
+
+    __slots__ = ("_dir_path", "_csv_kwargs", "_header_rows")
+
+    def __init__(self, dir_path: Path, **csv_kwargs):
+        super().__init__()
+        self._dir_path = Path(dir_path)
+        self._csv_kwargs: Dict[str, Any] = csv_kwargs
+        self._header_rows: List[int] = [0]
+
+    def load(self) -> TabularData:
+        """
+        Create a lazy loader for all CSV files in a directory and store them in a TabularData instance.
+        """
+
+        def lazy_csv_loader(csv_path: Path) -> Callable[[], pd.DataFrame]:
```

Comment:
Why do we have these two local functiond here (`lazy_csv_loader`/`csv_loader`) when we could just do the call of line 40 in line 46?

Reply:
It returns the bound function `csv_loader`, which is called later.